server.go

```diff
@@ -35,31 +41,34 @@ func (c Context) Check(msg string, err error) error {
 	return nil
 }
 
-// EmptyResonse writes a 204 No Content response. This should be used when the
-// client should not expect a response body.
-func (c Context) EmptyResonse() {
-	c.ResponseWriter.WriteHeader(http.StatusNoContent)
-}
-
-// Encode writes the JSON encoding of v to the response body.
-func (c Context) Encode(v interface{}) {
-	c.ResponseWriter.Header().Set("Content-Type", "application/json")
-	// encode nil slices as [] and nil maps as {} (instead of null)
-	if val := reflect.ValueOf(v); val.Kind() == reflect.Slice && val.Len() == 0 {
-		c.ResponseWriter.Write([]byte("[]\n"))
-		return
-	} else if val.Kind() == reflect.Map && val.Len() == 0 {
-		c.ResponseWriter.Write([]byte("{}\n"))
-		return
+// Encode writes the encoding of v to the response body. If v implements the
+// ResponseWriter interface, it is written directly.
+// Otherwise, it is marshalled as JSON.
+func (c Context) Encode(v any) {
+	switch v := v.(type) {
+	case nil:
+		c.ResponseWriter.WriteHeader(http.StatusNoContent)
+	case ResponseWriter:
```

Comment:
More of a question rather than feedback, but I assume the ResponseWriter interface is made to accomodate places like this?

```
func (b *Bus) slabsPartialHandlerGET(jc jape.Context) {
        jc.Custom(nil, []byte{})

        var key object.EncryptionKey
        if jc.DecodeParam("key", &key) != nil {
                return
        }

        var offset int
        if jc.DecodeForm("offset", &offset) != nil {
                return
        } else if offset < 0 {
                jc.Error(api.ErrInvalidOffset, http.StatusBadRequest)
                return
        }

        var length int
        if jc.DecodeForm("length", &length) != nil {
                return
        } else if length <= 0 {
                jc.Error(api.ErrInvalidLength, http.StatusBadRequest)
                return
        }

        data, err := b.ms.FetchPartialSlab(jc.Request.Context(), key, uint32(offset), uint32(length))
        if errors.Is(err, api.ErrObjectNotFound) {
                jc.Error(err, http.StatusNotFound)
                return
        } else if err != nil {
                jc.Error(err, http.StatusInternalServerError)
                return
        }
        jc.ResponseWriter.Write(data)
}
```

For Prometheus alias types a helper like writeResponse in hostd/renterd would probably still be required or there'd be a lot of code duplication.